array/sample.js

```diff
@@ -0,0 +1,21 @@
+if (!Array.prototype.sample) {
+  Object.defineProperty(Array.prototype, "sample", {
+    value: function(num){
+      var clone = this.slice(0);
+      var returnedArray = [];
+      var n = num ? num : 1;
+
+      if(n == 1){
+        return clone[Math.floor(Math.random()*clone.length)];
```

Comment:
There's no need to have an wlse statement here. If n == 1, the function will be exited.